memcache/memcache.go

```diff
@@ -188,14 +198,18 @@ func (cn *conn) condRelease(err *error) {
 	} else {
 		cn.nc.Close()
 	}
+	// Regardless of whether this connection is reusable, we should decrement the numActiveConns
+	// counter and then single the cond to wake up any goroutines waiting for a free active
+	// connection slot to open up.
+	cn.c.lk.Lock()
+	cn.c.numActiveConns[cn.addr.String()] -= 1
+	cn.c.lk.Unlock()
+	cn.c.cond.Signal()
 }
 
 func (c *Client) putFreeConn(addr net.Addr, cn *conn) {
 	c.lk.Lock()
 	defer c.lk.Unlock()
-	if c.freeconn == nil {
-		c.freeconn = make(map[string][]*conn)
-	}
```

Comment:
I moved this initialization up to the `NewFromSelector` function at https://github.com/tracelytics/gomemcache/pull/1/files#diff-789b976323c7e2b437c8c37b5674e7b8R132.

I guess that the base version of this code was written such that you could get away with instantiating your own `Client` without using the `New*` functions, but I added two more properties that need to be initialized for `Client`, and lazy initialization of all those things starts to get kind of verbose. Kind of a bikeshed change, but *shrug*... I'm not planning to push for this patch to be merged upstream, as that seems kind of intentionally barebones (also, while there are other forks, none have seemed to have the need to implement a max connection limit).